akka-stream/src/main/scala/akka/stream/io/OutputStreamSource.scala

```diff
@@ -0,0 +1,61 @@
+/**
+ * Copyright (C) 2015 Typesafe Inc. <http://www.typesafe.com>
+ */
+package akka.stream.io
+
+import java.io.OutputStream
+
+import akka.stream._
+import akka.stream.impl.io.OutputStreamSourceStage
+import akka.stream.scaladsl.{ FlowGraph, Source }
+import akka.util.ByteString
+
+import scala.concurrent.Future
+import scala.concurrent.duration._
+import scala.language.implicitConversions
+
+/**
+ * Source which allows to use [[OutputStream]] to interact with reactive stream.
+ */
+object OutputStreamSource {
+  import scala.language.postfixOps
+
+  /**
+   * Creates a synchronous (Java 6 compatible) Source.
+   *
+   * It materializes a [[Future]] containing the number of bytes written to OutputStream upon completion and
+   * [[OutputStream]] to interacting with reactive stream.
+   */
+  def apply(timeout: FiniteDuration = 5.seconds): Source[ByteString, OutputStream] = {
+    Source.apply2 { implicit b: FlowGraph.Builder[OutputStream] ⇒
```

Comment:
Since `OutputStreamSourceStage` has the right shape already, all you need is:

    Source.wrap(new OutputStreamSourceStage(...))